lib/puma/cluster.rb

```diff
@@ -536,9 +549,9 @@ def timeout_workers
       @workers.each do |w|
         if !w.term? && w.ping_timeout <= Time.now
           details = if w.booted?
-                      "(worker failed to check in within #{@options[:worker_timeout]} seconds)"
+                      "(Worker #{w.index} failed to check in within #{@options[:worker_timeout]} seconds)"
                     else
-                      "(worker failed to boot within #{@options[:worker_boot_timeout]} seconds)"
+                      "(Worker #{w.index} failed to boot within #{@options[:worker_boot_timeout]} seconds)"
```

Comment:
I'm happy to move this to a separate PR if deemed irrelevant. The reasoning here is that the boot logs include both the index and the pid, so I updated this for consistency. Logs before are visible here: https://github.com/puma/puma/issues/2786#issuecomment-1704766900.